### PR TITLE
Rename Bytes -> TextEncodingNone

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title.

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -55,22 +55,24 @@ Externals
     externals.cmath
     externals.libdevice
     externals.macros
-    externals.omniscidb
+    externals.heavydb
     externals.stdio
 
 
 OmniSciDB Backend
 =================
```
Comment (tupui) [inline doc/api.rst:63]:
```suggestion
HeavyDB Backend
===============
```

File: rbc/heavyai/text_encoding_dict.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-'''Omnisci Bytes type that corresponds to Omnisci type TEXT ENCODED NONE.
+'''Omnisci TextEncodingDict type that corresponds to Omnisci type TEXT ENCODED DICT.
```
Comment (tupui) [inline rbc/heavyai/text_encoding_dict.py:1]:
```suggestion
'''HeavyDB TextEncodingDict type that corresponds to Omnisci type TEXT ENCODED DICT.
```

File: rbc/heavyai/text_encoding_none.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+from .buffer import (
+    BufferPointer, Buffer, OmnisciBufferType,
+    omnisci_buffer_constructor)
+from numba.core import types, extending, cgutils
+from llvmlite import ir
+from typing import Union
+
+
+class HeavyDBTextEncodingNoneType(OmnisciBufferType):
+    """Omnisci TextEncodingNone type for RBC typesystem.
```
Comment (tupui) [inline rbc/heavyai/text_encoding_none.py:19]:
```suggestion
    """HeavyDB TextEncodingNone type for RBC typesystem.
```

File: rbc/tests/heavyai/test_text_encoding.py [hunk 4]
```diff
@@ -227,6 +227,7 @@ def test_text_encoding_column_list2(heavydb, size):
 @pytest.mark.usefixtures("create_columns")
 @pytest.mark.parametrize("size", (32,))
 @pytest.mark.parametrize("num_cols", (1, 2, 3, 4))
+@pytest.mark.skip()
```
Comment (tupui) [inline rbc/tests/heavyai/test_text_encoding.py:230]: Leftover? If not make sure to keep track of it somewhere.

File: rbc/heavyai/text_encoding_none.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    if isinstance(a, TextEncodingNonePointer):
+        if isinstance(b, (TextEncodingNonePointer, types.StringLiteral)):
+            def impl(a, b):
+                return not(a == b)
+            return impl
+
+
+@extending.lower_builtin(TextEncodingNone, types.Integer)
+def omnisci_text_encoding_none_constructor(context, builder, sig, args):
+    return omnisci_buffer_constructor(context, builder, sig, args)
```
Comment (tupui) [inline rbc/heavyai/text_encoding_none.py:122]:
```suggestion
def heavydb_text_encoding_none_constructor(context, builder, sig, args):
    return heavydb_buffer_constructor(context, builder, sig, args)
```

File: rbc/heavyai/text_encoding_none.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+
+@extending.lower_builtin(TextEncodingNone, types.StringLiteral)
+def omnisci_text_encoding_none_constructor_literal(context, builder, sig, args):
```
Comment (tupui) [inline rbc/heavyai/text_encoding_none.py:126]:
```suggestion
def heavydb_text_encoding_none_constructor_literal(context, builder, sig, args):
```

File: rbc/heavyai/text_encoding_none.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    int64_t = ir.IntType(64)
+    int8_t_ptr = ir.IntType(8).as_pointer()
+
+    literal_value = sig.args[0].literal_value
+    sz = int64_t(len(literal_value))
+
+    # arr = {ptr, size, is_null}*
+    arr = omnisci_buffer_constructor(context, builder, sig.return_type(types.int64), [sz])
```
Comment (tupui) [inline rbc/heavyai/text_encoding_none.py:134]:
```suggestion
    arr = heavydb_buffer_constructor(context, builder, sig.return_type(types.int64), [sz])
```

File: rbc/heavyai/text_encoding_none.py [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    msg_const = cgutils.make_bytearray(msg_bytes)
+    msg_global_var = cgutils.global_constant(builder.module, f"Text({literal_value})", msg_const)
+    msg_ptr = builder.bitcast(msg_global_var, int8_t_ptr)
+    sizeof_char = TargetInfo().sizeof('char')
+    cgutils.raw_memcpy(builder, ptr, msg_ptr, sz, sizeof_char)
+    return arr
+
+
+@extending.type_callable(TextEncodingNone)
+def type_omnisci_text_encoding_none(context):
```
Comment (tupui) [inline rbc/heavyai/text_encoding_none.py:147]:
```suggestion
def type_heavydb_text_encoding_none(context):
```

File: rbc/heavyai/text_encoding_none.py [hunk 9]
```diff
@@ -0,0 +1,153 @@
+'''Omnisci TextEncodingNone type that corresponds to Omnisci type TEXT ENCODED NONE.
```
Comment (tupui) [inline rbc/heavyai/text_encoding_none.py:1]:
```suggestion
'''HeavyDB TextEncodingNone type that corresponds to Omnisci type TEXT ENCODED NONE.
```

**Review by tupui — COMMENTED** [review]
Thanks @guilhermeleobas, I have a few suggestions.

File: rbc/tests/heavyai/test_text_encoding.py (hunk 4 above)
Comment (guilhermeleobas) [reply rbc/tests/heavyai/test_text_encoding.py:230]: There's a bug in HeavyDB master which crashes the server when running this test. I'll open an issue to investigate it.

Comment (tupui) [reply rbc/tests/heavyai/test_text_encoding.py:230]: ok thanks

**Review by pearu — COMMENTED** [review]
LGTM but requires resolving merge conflicts.

Thanks, @guilhermeleobas , for this update!

**Review by tupui — APPROVED** [review]
LGTM too now.

**Review by pearu — APPROVED** [review]
LGTM!

**Pull request merged**